flocker/acceptance/testtools.py

```diff
@@ -1166,155 +1010,6 @@ def configure_container_agent(node):
     return decorator
 
 
-def is_container_agent_running(node):
-    """
-    Check if the container agent is running on the specified node.
-
-    :param Node node: the node to check.
-    :return Deferred[bool]: a Deferred that will fire when
-        with whether the container agent is runnning.
-    """
-    d = node.run_script("service_running", "flocker-container-agent")
-
-    def not_existing(failure):
-        failure.trap(ProcessTerminated)
-        return False
-    d.addCallbacks(lambda result: True, not_existing)
-    return d
-
-
-def set_container_agent_enabled_on_node(node, enabled):
-    """
-    Ensure the container agent is enabled/disabled as specified.
-
-    :param Node node: the node on which to ensure the container
-        agent's state
-    :param bool enabled: True to ensure the container agent
-        is enabled and running, false to ensure the opposite.
-    :return Deferred[None]: a Deferred that will fire when
-        the container agent is in the desired state.
-    """
-    if enabled:
-        d = node.run_script("enable_service", "flocker-container-agent")
-    else:
-        d = node.run_script("disable_service", "flocker-container-agent")
-    # If the agent was disabled We have to reboot to clear the control cache.
-    # If we want to avoid the reboot we could add an API to do this.
-    if not enabled:
-        d.addCallback(lambda _: node.reboot())
-        # Wait for reboot to be far enough along that everything
-        # should be shutdown:
-        d.addCallback(lambda _: deferLater(reactor, 20, lambda: None))
-        # Wait until server is back up:
-        d = d.addCallback(lambda _:
-                          verify_socket(node.public_address, 22))
-        d.addCallback(lambda _: loop_until(
-            reactor, lambda: is_process_running(
-                node, b'flocker-dataset-agent')))
-        d.addCallback(
-            lambda _:
-            node.run_script("disable_service", "flocker-dataset-agent"))
-        d.addCallback(
-            lambda _:
-            node.run_script("enable_service", "flocker-dataset-agent"))
-        d.addCallback(lambda _: loop_until(
-            reactor, lambda: is_process_running(
-                node, b'flocker-dataset-agent')))
-    # Hide the value in the callback as it could come from
-    # different places and shouldn't be used.
-    d.addCallback(lambda _: None)
-    return d
-
-
-def is_process_running(node, name):
```

Comment:
That function is no longer used inside Flocker so I decided to remove it.
Same for a bunch of other functions.
It seemed better than leaving them lying around and future maintainers having to think about why the function exists.